src/utils.js

```
@@ -53,15 +54,26 @@ export const getShortContent = (content, length = 4) => (
 /**
  * Get amount text value and transform in its integer value
  *
- * "10" => 1000
- * "10.00" => 1000
- * "10,01" => 1001
- * "1000" => 100000
- * "1000.00" => 100000
+ * "10" => 1000n
+ * "10.00" => 1000n
+ * "10,01" => 1001n
+ * "1000" => 100000n
+ * "1000.00" => 100000n
+ *
+ * @return {BigInt} The integer value as a BigInt
  */
 export const getIntegerAmount = (value) => {
   const parsedValue = parseFloat(value.replace(',', '.'));
-  return Math.round(parsedValue * (10 ** hathorLib.constants.DECIMAL_PLACES));
+  const rawAmount = Math.round(parsedValue * (10 ** hathorLib.constants.DECIMAL_PLACES));
+
+  try {
+    // Convert to BigInt using the schema from wallet-lib
+    return bigIntCoercibleSchema.parse(rawAmount);
+  } catch (e) {
+    console.error('Failed to convert amount to BigInt:', e);
+    // Return 0n as fallback
+    return 0n;
```

Should we just let it crash?